weed/server/filer_grpc_server_rename.go

```diff
@@ -34,6 +34,20 @@ func (fs *FilerServer) AtomicRenameEntry(ctx context.Context, req *filer_pb.Atom
 		return nil, fmt.Errorf("%s/%s not found: %v", req.OldDirectory, req.OldName, err)
 	}
 
+	if oldEntry.IsDirectory() {
+		targetDir := newParent.Child(req.NewName)
+		newEntry, err := fs.filer.FindEntry(ctx, targetDir)
+		if err == nil {
+			if !newEntry.IsDirectory() {
+				fs.filer.RollbackTransaction(ctx)
+				return nil, fmt.Errorf("%s is not directory", targetDir)
+			}
+			if entries, _, _ := fs.filer.ListDirectoryEntries(context.Background(), targetDir, "", false, 1, "", "", ""); len(entries) > 0 {
+				return nil, fmt.Errorf("%s is not empty", targetDir)
```

Comment:
should it also roll back the transaction?

Reply:
I don't think it needs transaction rollback, it is only used as a condition to judge whether the move operation can be done.

Reply:
line 42 has the rollback. why that is needed?

Reply:
I refer to the logic of your previous StreamRenameEntry function. The FindEntry function has database operations, so transactions are used here.

Reply:
At the same time, I have a question: There are also database operations in the ListDirectoryEntries function, so should a rollback operation be added to the StreamRenameEntry function and other functions that call ListDirectoryEntries?

Reply:
you may want to send a separate PR for this change.

Reply:
I'm considering to do it. Do you think there is anything else that needs to be changed in this PR?